Projection plots (#1926)
openmc-dev/openmc: docs/source/usersguide/plots.rst

          program (Visit, ParaView, etc.) if the number of voxels is large (>10
          million or so).  Thus if you want an accurate picture that renders
          smoothly, consider using only one voxel in a certain direction.

----------------
Projection Plots
----------------

.. image:: ../_images/hexlat_anim.gif
   :width: 200px

The :class:`openmc.ProjectionPlot` class presents an alternative method
of producing 3D visualizations of OpenMC geometries. It was developed to
overcome the primary shortcoming of voxel plots, that an enormous number
of voxels must be employed to capture detailed geometric features.

> Projection plots do volume rendering on material or

Reviewer: ```suggestion
Projection plots perform volume rendering on material or
```

cell volumes, with colors specified in the same manner as slice plots.
This is done using the native ray tracing capabilities within OpenMC,
so any geometry in which particles successfully run without overlaps
or leaks will work with projection plots.

One drawback of projection plots is that particle tracks cannot be overlaid

> on them at present. Moreover, checking for overlap regions as is possible
> with slice plots is also not available. The image heading this section can

Reviewer: ```suggestion
on them at present. Moreover, checking for overlap regions is not currently possible with projection plots. The image heading this section can
```

      thisp.wireframe_regions = [fuel]
      thisp.wireframe_thickness = 2
  
      plot_file.append(thisp)

This generates a sequence of png files which can be joined to form a gif.
Each image specifies a different camera position using some simple periodic
functions to create a perfectly looped gif. :attr:`ProjectionPlot.look_at`
defines where the camera's centerline should point at.
:attr:`ProjectionPlot.camera_position` similarly defines where the camera
is situated in the universe level we seek to plot. The other settings
resemble those employed by :class:`openmc.Plot`, with the exception of
the :class:`ProjectionPlot.set_transparent` method and :attr:`ProjectionPlot.xs`
dictionary. These are used to control volume rendering of material
volumes. "xs" here stands for cross section, and it defines material
opacities in units of inverse centimeters. Setting this value to a
large number would make a material or cell opaque, and setting it to
zero makes a material transparent. Thus, the :class:`ProjectionPlot.set_transparent`
can be used to make all materials in the geometry transparent. From there,
individual material or cell opacities can be tuned to produce the

> desired result.

Reviewer: Does the `orthographic_width` attribute need to be mentioned here as well?

Author: New commit addresses this.
